Draft/discussion: Adding Dijkstra's algo specific doc

amcandio opened the pull request:
**What do people think about having an specific doc section for Dijkstra's?**

I think this could help discoverability of the shortest path docs, especially through search. From Google search data we can see that "Dijkstra" is a more popular search term than "shortest path".

https://trends.google.com/trends/explore?q=Dijkstra,shortest%20path&hl=en

<img width="1154" height="675" alt="image" src="https://github.com/user-attachments/assets/61e68d3b-4bcc-4d8a-b0ed-57faf3321b16" />

rossbar reviewed (commented):
In principal I think this is a good idea - Dijkstra's algorithm and shortest-path in general are very popular topics.

amcandio commented on doc/reference/algorithms/dijkstra.rst:
```diff
@@ -0,0 +1,140 @@
+
+The algorithm terminates when all nodes have been visited, or when the smallest tentative distance among unvisited nodes is infinity.
+
+Pseudocode
```
Yeah, good point. I just removed the section!

amcandio commented:
> In principal I think this is a good idea - Dijkstra's algorithm and shortest-path in general are very popular topics.

Alright, addressed your comment and marked it as ready for review!

dschult commented on doc/reference/algorithms/shortest_paths/dijkstra.rst:
```diff
@@ -0,0 +1,115 @@
+------------------
+Given a weighted graph G = (V, E) and a source vertex s ∈ V, compute the
+shortest-path distances d(s, v) from s to every vertex v ∈ V, where each
+edge (u, v) ∈ E has a non-negative weight w(u, v). Optionally, the algorithm
```
I think latex is currently preferred over unicode for the docs.
```suggestion
Given a weighted graph G = (V, E) and a source vertex $s \epsilon V$, compute the
shortest-path distances d(s, v) from s to every vertex $v \epsilon V$, where each
edge $(u, v) \epsilon E$ has a non-negative weight w(u, v). Optionally, the algorithm
```

dschult commented on doc/reference/algorithms/shortest_paths/dijkstra.rst:
```diff
@@ -0,0 +1,115 @@
+
+1. Assign every node a tentative distance value:
+    * `dist[s] = 0` for the source
+    * `dist[v] = ∞` for all other nodes
```
Is this initialization needed? Aren't all reachable nodes assigned a distance during the search?
This feels like an implementation detail.

Also, should these be double-ticks? Well, they are for code (I think that's right) so if you are thinking of this as code, the unicode infinity symbol shouldn't be there. If not code, maybe use latex for the part with `\infty`.

amcandio replied:
Will simplify and use LaTex, looks nicer.

dschult commented on doc/reference/algorithms/shortest_paths/dijkstra.rst:
```diff
@@ -0,0 +1,115 @@
+1. Assign every node a tentative distance value:
+    * `dist[s] = 0` for the source
+    * `dist[v] = ∞` for all other nodes
+2. Initialize all nodes as unvisited.
```
Similarly, do we need to mark them as unvisited? Or just not mark them as visited? Why do we include this step in the description? People trying to understand it maybe don't need to pay attention to this.

amcandio commented:
Pushed a second revision with comments addressed and some rework!

amcandio commented:
Ah didn't realize that! I simplified and also fixed some spacing that was causing some odd rendering in Sphinx

rossbar reviewed (approved):
Thanks @amcandio , LGTM!

I took the liberty of pushing up a few rst/sphinx-y tweaks, mostly to handle doc build warnings and use built-in roles where possible. One thing I did directly undoes one of @dschult 's suggestions: I had to switch the `$` math back to the `:math:` role in the complexity section. The reason for doing so is that the dollar math extension we use hooks into the doc build *after* the docstring has already been parsed by sphinx, and sphinx treats vertical bars (i.e. `|`) as characters for substitution. Anyways - the details are arcane, but *in general* I agree with @dschult that `$` math reads better, but for this one corner case where `|` characters are involved!
